pkg/network/discovery.go

```diff
@@ -105,6 +114,28 @@ func (d *DefaultDiscovery) BadPeers() []string {
 	return addrs
 }
 
+// GoodPeers returns all addresses of known good peers (that at least once
+// succeded handshaking with us.
+func (d *DefaultDiscovery) GoodPeers() []string {
```

Comment:
GoodPeers == ConnectedPeers? or not?

Reply:
not yet got it. Just only handshaked

Reply:
Not exactly, semantically it's `ConnectedOnce`, they can be disconnected now, but we were able to do a complete handshake with them once upon a time, so they're probably good.